components/gitpod-db/go/personal_access_token.go

```diff
@@ -87,6 +87,30 @@ func CreatePersonalAccessToken(ctx context.Context, conn *gorm.DB, req PersonalA
 	return token, nil
 }
 
+func DeletePersonalAccessTokenForUser(ctx context.Context, conn *gorm.DB, tokenID uuid.UUID, userID uuid.UUID) (int64, error) {
+	if tokenID == uuid.Nil {
+		return 0, fmt.Errorf("Invalid or empty tokenID")
+	}
+
+	if userID == uuid.Nil {
+		return 0, fmt.Errorf("Invalid or empty userID")
+	}
+
+	db := conn.WithContext(ctx)
+
+	db = db.
+		Table((&PersonalAccessToken{}).TableName()).
+		Where("id = ?", tokenID).
+		Where("userId = ?", userID).
+		Where("deleted = ?", 0).
+		Update("deleted", 1)
+	if db.Error != nil {
+		return 0, fmt.Errorf("Failed to delete token: %v", db.Error)
```

Comment:
[nit] To follow Go recommended style. 
```suggestion
		return 0, fmt.Errorf("failed to delete token (ID: %s): %v", tokenID.String(), db.Error)
```
We do use first letter uppercase in the API Handlers to provide a cleaner user-facing error message, but here in the `db` package we can stick to standard Go practice.

Here, the suggestion also includes the Token ID in the error. This makes tracking down the entry a bit simpler and I'd generally recommend including non sensitive IDs in the errors like this

There are a couple more occurrences of the uppercase.

Reply:
We have lots of params checkings like code below, they are uppercase too, should we change them all? If so, it could be a follow-up PR. cc @easyCZ 

```
if req.UserID == uuid.Nil {
	return PersonalAccessToken{}, fmt.Errorf("Invalid or empty userID")
}
```

Reply:
So the only place where we "should" use Uppercase is the `apiv1` handler code. As that's what ultimately returns a response to the user. Everywhere else `errors` should start with lowercase as per standard go style. Sorry, this is a bit confusing. Maybe I can write a middleware which uppercases the first letter of the error to get rid of this inconcistency.